### ISSUE TITLE
[UNet_Medical/TensorFlow2] Training accuracy not reproducible on V100

### ISSUE THREAD

**Issue by ClementPinard** [NVIDIA/DeepLearningExamples#818]
Related to **UNet_Medical/TensorFlow2** 

**Describe the bug**
Hi, I have 1 V100 GPU, so I ran the examples script, trying to reproduce the results reported in [this table](https://github.com/NVIDIA/DeepLearningExamples/tree/master/TensorFlow2/Segmentation/UNet_Medical#training-accuracy-nvidia-dgx-1-8x-v100-16gb), the row with 1 GPU.
Unfortunately, I get a lower Dice score than you. Interestingly enough, It's even lower for FP32, not sure what makes the model less accurate with FP32.

My reported results for Dice score
| Precision | Fold0 | Fold1 | Fold2 | Fold3 | Fold4 | Mean | README |
|:-------------| -------:| --------:| -------:| -------:| --------:| --------:| ------------: |
| FP32       | 0.845 | 0.844 | 0.832 | 0.835 | 0.850 | 0.841 | 0.8901 |
|Mixed       | 0.870 | 0.859 | 0.854 | 0.862 | 0.866 | 0.862 | 0.8898 |

I also tried to plot the training dice loss, and you can see that they are very different from FP32 to AMP, FP32 seems to suffer a lot from instability, is this normal ? (On the contrary, the cross entropy losses are very similar)

![Figure_1](https://user-images.githubusercontent.com/4380424/106185310-f3ee2980-61a2-11eb-818f-51c8fc4ead52.png)

- I only plotted fold0 curves, but the behavior is the same throughout all the folds
- Also notice that the Dice loss does not go below 0.1 which means even training dice score is lower that reported validation dice score.

**To Reproduce**
1. Launch the scripts `examples/unet_TRAIN_1GPU.sh` and `examples/unet_TRAIN_TF_AMP_1GPU.sh`
**Expected behavior**
Dice score similar to what is reported on the README with a single V100

**Environment**
Please provide at least:
* Container version (e.g. pytorch:19.05-py3): tensorflow:20.06-tf2-py3
* GPUs in the system: (e.g. 8x Tesla V100-SXM2-16GB): 1x Tesla V100-SXM2-16GB
* CUDA driver version (e.g. 418.67): 450.51.05

**Comment by mmarcinkiewicz** [comment]
Thanks @ClementPinard , I'll investigate the issue.

**Comment by mmarcinkiewicz** [comment]
@ClementPinard in the meantime, could you provide a log of the whole training?

**Comment by ClementPinard** [comment]
Hi, thanks for your answer. Please find attached the training log : https://drive.google.com/file/d/19iVzCQyFnhDvB98-sX2hHW2MMPua7RYE/view?usp=sharing

Notice that I changed the bash script a little : 
https://github.com/NVIDIA/DeepLearningExamples/blob/master/TensorFlow2/Segmentation/UNet_Medical/examples/unet_TRAIN_1GPU.sh#L24
Here, I call the script in `runtime/parse_results.py` instead of `utils/parse_results.py` , I guess the folder got renamed at some point.

**Comment by mmarcinkiewicz** [comment]
There is something suspicious happening with your CE loss at the beginning of the training:
Yours:
[1,0]<stdout>:DLL 2021-01-27 15:03:59.798912 - (0, 6400) train_ce_loss : 7.529220104217529  train_dice_loss : 0.37845203280448914  train_total_loss : 7.907671928405762 
[1,0]<stdout>:DLL 2021-01-27 15:04:50.996539 - (100, 6400) train_ce_loss : 0.48795291781425476  train_dice_loss : 0.3218422532081604  train_total_loss : 0.8097951412200928 
[1,0]<stdout>:DLL 2021-01-27 15:05:36.643161 - (200, 6400) train_ce_loss : 0.48255500197410583  train_dice_loss : 0.28111955523490906  train_total_loss : 0.7636745572090149

Mine:
[1,0]<stdout>:DLL 2021-02-02 08:37:19.824353 - (0, 6400) train_ce_loss : 0.6914711594581604  train_dice_loss : 0.3778577148914337  train_total_loss : 1.0693289041519165 
[1,0]<stdout>:DLL 2021-02-02 08:37:40.646545 - (100, 6400) train_ce_loss : 0.34419772028923035  train_dice_loss : 0.19065582752227783  train_total_loss : 0.5348535776138306 
[1,0]<stdout>:DLL 2021-02-02 08:37:54.409618 - (200, 6400) train_ce_loss : 0.3495800495147705  train_dice_loss : 0.17876121401786804  train_total_loss : 0.5283412933349609 

Can you try to download the data again, maybe it got corrupted along the way?

**Comment by mmarcinkiewicz** [comment]
Other things to try is to:
1. Pull the latest container instead of 20.06, meaning replace 20.06 with 20.12 in [here](https://github.com/NVIDIA/DeepLearningExamples/blob/master/TensorFlow2/Segmentation/UNet_Medical/Dockerfile#L1)
2. Make sure you don't install any other libraries in the container which may influence the TF version
3. Play with assigning a lower weight to the CE loss by multiplying it by `0.1` for example.

**Comment by ClementPinard** [comment]
- Just had a check on the 30 train images, and they all seem fine.
 - i'll try to run a new experiment with latest docker image.
 - I used the docker image as is without installing anything
 - I'll try with another wieght for CE loss.
 
 in the mean time do you have the exact commit the results were made with ?

**Comment by mmarcinkiewicz** [comment]
Ok, found it. @ClementPinard could you please replace the whole [CE Loss](https://github.com/NVIDIA/DeepLearningExamples/blob/master/TensorFlow2/Segmentation/UNet_Medical/runtime/losses.py#L35) with

```python
crossentropy_loss = tf.reduce_mean(input_tensor=tf.nn.softmax_cross_entropy_with_logits(logits=flat_logits,
                                                                                            labels=flat_labels),
                                       name='cross_loss_ref')
```

Please let me know if that solves the issue.

**Comment by mmarcinkiewicz** [comment]
@ClementPinard the code was updated (including your fix to `utils/parse_results.py` -> `runtime/parse_results.py`). If the issue still persists please reopen.
